api/server/router/container/container_routes.go

```diff
@@ -137,7 +137,7 @@ func (s *containerRouter) postContainersStart(ctx context.Context, w http.Respon
 	// A non-nil json object is at least 7 characters.
 	if r.ContentLength > 7 || r.ContentLength == -1 {
 		if versions.GreaterThanOrEqualTo(version, "1.24") {
-			return validationError{fmt.Errorf("starting container with HostConfig was deprecated since v1.10 and removed in v1.12")}
+			return validationError{fmt.Errorf("starting container with any parameters was deprecated since v1.10 and removed in v1.12")}
```

Comment:
Historically, it only accepted a HostConfig as body; https://docs.docker.com/v1.4/reference/api/docker_remote_api_v1.11/#start-a-container, and always referenced it as `HostConfig` in the deprecation message; https://docs.docker.com/engine/reference/api/docker_remote_api_v1.18/#/start-a-container

Perhaps we should change it to be clearer that this is about the *body* (i.e. not query parameters)